E0137 needs to be updated to new format

From: src/test/compile-fail/E0137.rs

Error E0137 needs a span_label, updating it from:

```
error[E0137]: multiple functions with a #[main] attribute
  --> src/test/compile-fail/E0137.rs:17:1
   |
17 | fn f() {} //~ ERROR E0137
   | ^^^^^^^^^
```

To:

```
error[E0137]: multiple functions with a #[main] attribute
  --> src/test/compile-fail/E0137.rs:17:1
   |
17 | fn f() {} //~ ERROR E0137
   | ^^^^^^^^^ additional #[main] function
```

Bonus: add the span_label to show where the first duplicate was created:

```
error[E0137]: multiple functions with a #[main] attribute
  --> src/test/compile-fail/E0137.rs:17:1
   |
14 | fn foo() {}
   | ----------- first #[main] function
17 | fn f() {} //~ ERROR E0137
   | ^^^^^^^^^ additional #[main] function
```

--- a/src/librustc/middle/entry.rs
+++ b/src/librustc/middle/entry.rs
@@ -121,8 +121,11 @@ fn find_item(item: &Item, ctxt: &mut EntryContext, at_root: bool) {
             if ctxt.attr_main_fn.is_none() {
                 ctxt.attr_main_fn = Some((item.id, item.span));
             } else {
-                span_err!(ctxt.session, item.span, E0137,
-                          "multiple functions with a #[main] attribute");
+                struct_span_err!(ctxt.session, item.span, E0137,
+                          "multiple functions with a #[main] attribute")
+                .span_label(item.span, &format!("additional #[main] function"))
+                .span_label(ctxt.attr_main_fn.unwrap().1, &format!("first #[main] function"))
+                .emit();
             }
         },
         EntryPointType::Start => {

--- a/src/test/compile-fail/E0137.rs
+++ b/src/test/compile-fail/E0137.rs
@@ -11,7 +11,9 @@
 #![feature(main)]
 
 #[main]
-fn foo() {}
+fn foo() {} //~ NOTE first #[main] function
 
 #[main]
-fn f() {} //~ ERROR E0137
+fn f() {}
+//~^ ERROR E0137
+//~| NOTE additional #[main] function

